### PR TITLE
REF: reorganize pandas/tests/test_series.py

### PR DESCRIPTION
**Pull request by wesm**
Similar to test_frame.py, but only 8K lines to deal with this time.

### CODE REVIEW

**Comment by wesm** [comment]
The sparse tests were relying on a small subset of unit tests in `CheckNameIntegration` (including several tests which didn't use non-sparse Series at all). I'll fix this up in the next day or two to get the build passing so this doesn't linger too long

**Comment by wesm** [comment]
@jreback Think I fixed the sparse tests, merge away after review when ready

**Comment by jreback** [comment]
thanks!

File: pandas/tests/series/common.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @cache_readonly
+    def ts(self):
+        ts = _ts.copy()
+        ts.name = 'ts'
+        return ts
+
+    @cache_readonly
+    def series(self):
+        series = tm.makeStringSeries()
+        series.name = 'series'
```
Comment (jreback) [inline pandas/tests/series/common.py]: not directly related, and maybe I am missing something, but these have to be strictly read only, enforced by the user. IOW. If in a single test class I modify one of these (say set an element of the Series), won't that persist?

Comment (wesm) [inline pandas/tests/series/common.py]: Oops, good point; to be on the safe side we should clear() the `_cache` variable (set by `cache_readonly`) in between unit tests. same with the DataFrame tests

**Pull request closed without merging**